lerobot/common/robot_devices/utils.py

```diff
@@ -1,19 +1,37 @@
-import platform
 import time
 
 
-def busy_wait(seconds):
-    if platform.system() == "Darwin":
-        # On Mac, `time.sleep` is not accurate and we need to use this while loop trick,
-        # but it consumes CPU cycles.
-        # TODO(rcadene): find an alternative: from python 11, time.sleep is precise
-        end_time = time.perf_counter() + seconds
-        while time.perf_counter() < end_time:
+def precise_sleep(seconds: float, blocking: bool = False):
+    """A more precise sleep than time.sleep
+
+    There are several factors that influence the precision of time.sleep. They basically boil down to the OS
+    timer granularity (see https://docs.python.org/3.11/whatsnew/3.11.html#time), and the system load which
+    influences CPU scheduling.
+
+    In all cases, nothing much can be done about issues with CPU scheduling. To be as precise as possible,
+    we need to use a while-loop with a `pass`. This unfortunately keeps the CPU busy the whole time.
+
+    The trick we use here, is to sleep most of the requested time, but switch to blocking the CPU towards the
+    end of the requested time.
+
+    NOTE: See the benchmarking script at the bottom of the source file.
+
+    Args:
+        seconds: Amount of time to sleep for in seconds.
+        blocking: If set to True, just does a while-loop with `pass` making the sleep as precise as possible
+            but using the CPU the whole time. Set this to True, if you don't need to parallelize with other
+            work.
+    """
+    end_time = time.perf_counter() + seconds
+    # 20 ms buffer is usually enough to account for an overly-long last sleep, even with heavy system load.
+    end_time_with_buffer = end_time - 0.02
```

Comment:
I tried increasing this buffer till I saw clear advantage in using `precise_sleep` instead of `time.sleep` with maximal system load. Please check on your end.